---
title: "Exercise ChIP-seq"
author: "Mark Ziemann"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_float: true
    fig_width: 5
    fig_height: 5
theme: cosmo
---

## Introduction

```{r,packages}

suppressPackageStartupMessages({
    library("RhpcBLASctl")
    library("reshape2")
    library("DESeq2")
    library("gplots")
    library("mitch")
    library("eulerr")
    library("limma")
    library("topconfects")
    library("beeswarm")
    library("kableExtra")
})

RhpcBLASctl::blas_set_num_threads(1)

```

## Import read counts

Rename samples. Don't use initials.

```{r,importdata}

gt <- read.table("gt.tsv")
gt$ens <- sapply(strsplit(gt[,1]," "),"[[",1)
dim(gt)

k9 <- read.table("raw_data/h3k9ac/h3k9_counts.tsv",header=TRUE)
dim(k9)

k9a <- merge(k9,gt,by.x="Geneid",by.y="ens",all.x=TRUE)

rownames(k9a) <- paste(k9a$Geneid,k9a$genesymbol,k9a$Chr,k9a$Start,k9a$End)

k9a <- k9a[,7:18]
colnames(k9a) <- c("pos1","pre1","pos2","pre2","pos3","pre3","pos4","pre4","pos5","pre5","pos6","pre6")

head(k9a) |>
  kbl(caption = "H3K9ac TSS data") |>
  kable_paper("hover", full_width = F)

colSums(k9a)

k36 <- read.table("raw_data/h3k36ac/h3k36_counts.tsv",header=TRUE)
dim(k36)

k36a <- merge(k36,gt,by.x="Geneid",by.y="ens",all.x=TRUE)
rownames(k36a) <- paste(k36a$Geneid,k36a$genesymbol,k36a$Chr,k36a$Start,k36a$End)

k36a <- k36a[,7:18]
colnames(k36a) <- c("pos1","pre1","pos2","pre2","pos3","pre3","pos4","pre4","pos5","pre5","pos6","pre6")

head(k36a) |>
  kbl(caption = "H3K36ac TSS data") |>
  kable_paper("hover", full_width = F)

colSums(k36a)

rpm9 <- apply(k9a, 2 , function(x) { x / sum(x) } ) * 1000000
rpm9 <- rpm9[rowMeans(rpm9) > 1,]

rpm36 <- apply(k36a, 2 , function(x) { x / sum(x) } ) * 1000000
rpm36 <- rpm36[rowMeans(rpm36) > 1,]

```

## MDS Plot

```{r,mds1}

mds9 <- cmdscale(dist(t(k9a)))
cols=rep(c("pink","lightblue"),6)
XMIN=min(mds9[,1])*1.1
XMAX=max(mds9[,1])*1.1
plot(mds9, xlab="Coordinate 1", ylab="Coordinate 2", cex=2,col=cols,pch=19,main="H3K9ac",xlim=c(XMIN,XMAX),bty="n")
text(mds9, labels=colnames(k9a) ,cex=1.1)

pdf("k9a_mds.pdf",width=5,height=5)
plot(mds9, xlab="Coordinate 1", ylab="Coordinate 2", cex=2,col=cols,pch=19,main="H3K9ac",xlim=c(XMIN,XMAX),bty="n")
text(mds9, labels=colnames(k9a) ,cex=1.1)
dev.off()

mds36 <- cmdscale(dist(t(k36a)))
cols=rep(c("pink","lightblue"),6)
XMIN=min(mds36[,1])*1.1
XMAX=max(mds36[,1])*1.1
plot(mds36, xlab="Coordinate 1", ylab="Coordinate 2", cex=2,col=cols,pch=19,main="H3K36ac",xlim=c(XMIN,XMAX),bty="n")
text(mds36, labels=colnames(k9a) ,cex=1.1)

pdf("k36a_mds.pdf",width=5,height=5)
plot(mds36, xlab="Coordinate 1", ylab="Coordinate 2", cex=2,col=cols,pch=19,main="H3K36ac",xlim=c(XMIN,XMAX),bty="n")
text(mds36, labels=colnames(k9a) ,cex=1.1)
dev.off()

allac <- cbind(k9a,k36a)
mycolnames <- colnames(allac)
colnames(allac) <- paste(c(rep("k9",12),rep("k36",12)),mycolnames)

mdsa <- cmdscale(dist(t(allac)))
cols=rep(c("pink","lightblue"),6)
XMIN=min(mdsa[,1])*1.1
XMAX=max(mdsa[,1])*1.1
plot(mdsa, xlab="Coordinate 1", ylab="Coordinate 2", cex=2,col=cols,pch=19,main="ChIP",xlim=c(XMIN,XMAX),bty="n")
text(mdsa, labels=colnames(allac) ,cex=1.1)

```

## DESeq2

Sample sheet then differential analysis.

Remove genes with less than 10 reads per sample on average.

```{r,deseq_k9}

ss <- as.data.frame(colnames(k9a))
ss$pos <- factor(grepl("pos",ss[,1]))
ss$participant <- c("1","1","2","2","3","3","4","4","5","5","6","6")

dim(k9a)
k9a <- k9a[rowMeans(k9a)>=10,]
dim(k9a)

dds <- DESeqDataSetFromMatrix(countData = k9a , colData = ss, design = ~ participant + pos )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <- cbind(as.data.frame(z),assay(vsd))
dk9 <- as.data.frame(zz[order(zz$pvalue),])
write.table(dk9,file="k9a_deseq.tsv",quote=F,sep="\t")

head(dk9) |>
  kbl(caption = "Top significant genes in paired analysis for H3K9ac)") |>
  kable_paper("hover", full_width = F)

nrow(subset(dk9,padj<0.05 & log2FoldChange > 0))
nrow(subset(dk9,padj<0.05 & log2FoldChange < 0))

```

Smear plot

```{r,smearplot9}

sig9 <- subset(dk9,padj<0.05)
NSIG=nrow(sig9)
NDOWN=nrow(subset(sig9,log2FoldChange<0))
NUP=nrow(subset(sig9,log2FoldChange>0))
NTOT=nrow(dk9)
HEADER=paste(NTOT,"TSSs detected;",NSIG,"@5%FDR;",NUP,"up",NDOWN,"down")
plot(log10(dk9$baseMean),dk9$log2FoldChange,cex=0.5,pch=19,col="darkgray",
  xlab="log10 basemean",ylab="log2 fold change",main="H3K9ac")
points(log10(sig9$baseMean),sig9$log2FoldChange,cex=0.5,pch=19,col="black")
mtext(HEADER)

pdf("k9a_smear.pdf",width=5,height=5)
plot(log10(dk9$baseMean),dk9$log2FoldChange,cex=0.5,pch=19,col="darkgray",
  xlab="log10 basemean",ylab="log2 fold change",main="H3K9ac")
points(log10(sig9$baseMean),sig9$log2FoldChange,cex=0.5,pch=19,col="black")
mtext(HEADER)
dev.off()

```

Now for H3K36a

```{r,deseq_k36}

dim(k36a)
k36a <- k36a[rowMeans(k36a)>=10,]
dim(k36a)

dds <- DESeqDataSetFromMatrix(countData = k36a , colData = ss, design = ~ participant + pos )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <- cbind(as.data.frame(z),assay(vsd))
dk36 <- as.data.frame(zz[order(zz$pvalue),])
write.table(dk36,file="k36a_deseq.tsv",quote=F,sep="\t")

head(dk36) |>
  kbl(caption = "Top significant genes in paired analysis for H3K36ac)") |>
  kable_paper("hover", full_width = F)

nrow(subset(dk36,padj<0.05 & log2FoldChange > 0))
nrow(subset(dk36,padj<0.05 & log2FoldChange < 0))

```

Smear plot

```{r,smearplot36}

sig36 <- subset(dk36,padj<0.05)
NSIG=nrow(sig36)
NDOWN=nrow(subset(sig36,log2FoldChange<0))
NUP=nrow(subset(sig36,log2FoldChange>0))
NTOT=nrow(dk36)
HEADER=paste(NTOT,"TSSs detected;",NSIG,"@5%FDR;",NUP,"up",NDOWN,"down")
plot(log10(dk36$baseMean),dk36$log2FoldChange,cex=0.5,pch=19,col="darkgray",
  xlab="log10 basemean",ylab="log2 fold change",main="H3K9ac")
points(log10(sig36$baseMean),sig36$log2FoldChange,cex=0.5,pch=19,col="black")
mtext(HEADER)

pdf("k36a_smear.pdf",width=5,height=5)
plot(log10(dk36$baseMean),dk36$log2FoldChange,cex=0.5,pch=19,col="darkgray",
  xlab="log10 basemean",ylab="log2 fold change",main="H3K9ac")
points(log10(sig36$baseMean),sig36$log2FoldChange,cex=0.5,pch=19,col="black")
mtext(HEADER)
dev.off()

```

## Mitch

H3K9ac

```{r,mitch9,fig.height=7,fig.width=7}

pw <- gmt_import("ref/c5.go.v2024.1.Hs.symbols.gmt")

names(pw) <- gsub("_", " ", names(pw))

gt2 <- as.data.frame(rownames(k9a))

gt2$gene <- sapply(strsplit(gt2[,1]," "),"[[",2)

m9 <- mitch_import(x=dk9,DEtype="deseq2",geneTable=gt2)
head(m9)

mres9 <- mitch_calc(x=m9,genesets=pw,minsetsize=5,cores=8,priority="effect")

head(subset(mres9$enrichment_result,p.adjustANOVA<0.05 & s.dist>0),50) |>
  kbl(caption = "Top upregulated GO terms based on effect size (FDR<0.05) for H3K9ac") |>
  kable_paper("hover", full_width = F)

head(subset(mres9$enrichment_result,p.adjustANOVA<0.05 & s.dist<0),50) |>
  kbl(caption = "Top downregulated GO terms based on effect size (FDR<0.05) for H3K9ac") |>
  kable_paper("hover", full_width = F)

par(mar=c(5,27,3,3))
top <- mres9$enrichment_result
top <- subset(top,p.adjustANOVA<0.05)
nrow(top)
up <- head(subset(top,s.dist>0),20)
dn <- head(subset(top,s.dist<0),20)
top <- rbind(up,dn)
vec=top$s.dist
names(vec)=top$set
names(vec) <- gsub("_"," ",names(vec))
vec <- vec[order(vec)]
barplot(abs(vec),col=sign(-vec)+3,horiz=TRUE,las=1,cex.names=0.65,main="H3K9ac",xlab="ES")
grid()

pdf("k9a_mitchbar.pdf",width=7,height=7)
barplot(abs(vec),col=sign(-vec)+3,horiz=TRUE,las=1,cex.names=0.65,main="H3K9ac",xlab="ES")
grid()
dev.off()

par(mar=c(5.1, 4.1, 4.1, 2.1))

```

H3K36ac

```{r,mitch36,fig.height=7,fig.width=7}

m36 <- mitch_import(x=dk36,DEtype="deseq2",geneTable=gt2)
head(m36)

mres36 <- mitch_calc(x=m36,genesets=pw,minsetsize=5,cores=8,priority="effect")

head(subset(mres36$enrichment_result,p.adjustANOVA<0.05 & s.dist>0),50) |>
  kbl(caption = "Top upregulated GO terms based on effect size (FDR<0.05) for H3K36ac") |>
  kable_paper("hover", full_width = F)

head(subset(mres9$enrichment_result,p.adjustANOVA<0.05 & s.dist<0),50) |>
  kbl(caption = "Top downregulated GO terms based on effect size (FDR<0.05) for H3K36ac") |>
  kable_paper("hover", full_width = F)

par(mar=c(5,27,3,3))
top <- mres36$enrichment_result
top <- subset(top,p.adjustANOVA<0.05)
nrow(top)
up <- head(subset(top,s.dist>0),20)
dn <- head(subset(top,s.dist<0),20)
top <- rbind(up,dn)
vec=top$s.dist
names(vec)=top$set
names(vec) <- gsub("_"," ",names(vec))
vec <- vec[order(vec)]
barplot(abs(vec),col=sign(-vec)+3,horiz=TRUE,las=1,cex.names=0.65,main="H3K36ac",xlab="ES")
grid()

pdf("k36a_mitchbar.pdf",width=7,height=7)
barplot(abs(vec),col=sign(-vec)+3,horiz=TRUE,las=1,cex.names=0.65,main="H3K36ac",xlab="ES")
grid()
dev.off()

par(mar=c(5.1, 4.1, 4.1, 2.1))

```

## SLC

```{r,slc1}

rpm9_SLC2A4 <- rpm9[grep("SLC2A4",rownames(rpm9)),]
rpm9_SLC2A4
rpm9_SLC2A4 <- colSums(rpm9_SLC2A4)
rpm9_SLC2A4 <- matrix(rpm9_SLC2A4,nrow=2)
colnames(rpm9_SLC2A4) <- 1:6
rownames(rpm9_SLC2A4) <- c("Post","Pre")
rpm9_SLC2A4 <- rpm9_SLC2A4[c(2,1),]
barplot(rpm9_SLC2A4,beside=TRUE,col=c("#0047AB", "#D2042D"),main="SLC2A4 H3K9ac",ylab="counts per million",xlab="participant",ylim=c(0,73))
legend(11.5,73, legend=c("Pre", "Post"), fill=c("#0047AB", "#D2042D"), cex=1)

pdf("k9a_SLC2A4.pdf",width=5,height=5)
barplot(rpm9_SLC2A4,beside=TRUE,col=c("#0047AB", "#D2042D"),main="SLC2A4 H3K9ac",ylab="counts per million",xlab="participant",ylim=c(0,73))
legend(11.5,73, legend=c("Pre", "Post"), fill=c("#0047AB", "#D2042D"), cex=1)
dev.off()


rpm36_SLC2A4 <- rpm36[grep("SLC2A4",rownames(rpm36)),]
rpm36_SLC2A4
rpm36_SLC2A4 <- colSums(rpm36_SLC2A4)
rpm36_SLC2A4 <- matrix(rpm36_SLC2A4,nrow=2)
colnames(rpm36_SLC2A4) <- 1:6
rownames(rpm36_SLC2A4) <- c("Post","Pre")
rpm36_SLC2A4 <- rpm36_SLC2A4[c(2,1),]
barplot(rpm36_SLC2A4,beside=TRUE,col=c("#0047AB", "#D2042D"),main="SLC2A4 H3K36ac",ylab="counts per million",xlab="participant",ylim=c(0,70))
legend(3.5,70, legend=c("Pre", "Post"), fill=c("#0047AB","#D2042D"), cex=1)

pdf("k36a_SLC2A4.pdf",width=5,height=5)
barplot(rpm36_SLC2A4,beside=TRUE,col=c("#0047AB", "#D2042D"),main="SLC2A4 H3K36ac",ylab="counts per million",xlab="participant",ylim=c(0,70))
legend(3.5,70, legend=c("Pre", "Post"), fill=c("#0047AB","#D2042D"), cex=1)
dev.off()

```

## Session

```{r,session}

sessionInfo()

```
